### ISSUE TITLE
Question about ID Retrieval Evaluation and Dataset settings

### ISSUE THREAD

**Issue by dsj320** [Sanoojan/REFace#22]
Hi, thank you for your impressive work on REFace.

While studying the evaluation code to reproduce the ID retrieval metrics, I noticed a specific behavior regarding how images are preprocessed before being fed into ArcFace.

Observation In the evaluation script (e.g., for CelebA), the command uses --dataset "ffhq":

```

echo "ID similarity with Source using Arcface:" >> "$output_filename"
CUDA_VISIBLE_DEVICES=${device} python eval_tool/ID_retrieval/ID_retrieval.py --device cuda \
    "${source_path}" \
    "${Results_out}" \
    "${source_mask_path}" \
    "${target_mask_path}" \
    --dataset "ffhq" \  <-- Note here
    --print_sim True  \
    --arcface True >> "$output_filename" 
```
However, inside eval_tool/ID_retrieval/ID_retrieval.py (class MaskedImagePathDataset), the ffhq setting triggers a specific preserve list that masks out significant regions (likely hair, neck, background, etc.):

```

        if self.data_name=="celeba":
            preserve = [1,2,4,5,8,9 ,6,7,10,11,12 ]
        elif self.data_name=="ffhq":
            # This is triggered by the command above
            preserve = [1,2,3,5,6,7,9] 
        elif self.data_name=="ff++":
            preserve = [1,2,4,5,8,9 ]
        else:
            # Full image
            preserve=[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]

```

<img width="1249" height="538" alt="Image" src="https://github.com/user-attachments/assets/cac84dc8-5881-4969-a2ec-18ce7fb0585d" />

in the picture we can see that we don't preserve all of the faces,i think this way we can't get right id similarity and retrieval val,could  you please explain it?

Looking forward to your reply. Thanks!

**Comment by Sanoojan** [comment]
I guess this will give you more insights: https://github.com/Sanoojan/REFace/issues/18#issuecomment-2988038583

**Comment by dsj320** [comment]
> I guess this will give you more insights: [#18 (comment)](https://github.com/Sanoojan/REFace/issues/18#issuecomment-2988038583)

Hi authors,

Thank you for sharing such an impressive work and open-sourcing the code! I am currently studying your paper and attempting to reproduce the results based on this repository.

I have a couple of questions regarding the specific training details that would greatly help with my reproduction:

Training Strategy (Multi-stage vs. Single-stage): Does the training process involve multiple stages? For example, did you pre-train the adapters/projectors (alignment stage) while freezing the UNet first, and then fine-tune the whole model? Or was it trained end-to-end in a single stage from the beginning? I am asking because I noticed that training from scratch (or with standard initialization) sometimes leads to loss oscillation or difficulty in convergence.

Configuration Consistency: Is the training configuration provided in this repository (e.g., learning rate, loss weights in config files) the exact same configuration used to produce the results in the paper? I want to ensure I haven't missed any specific hyperparameters that are crucial for stability.

Any insights or guidance on the training pipeline would be highly appreciated.

Thanks again for your contribution!
